### DISCUSSION TITLE
About ProtoProps (stackTraceLimit and prepareStackTrace) of NativeError

### DISCUSSION THREAD

**Discussion by SuperMonster003** [mozilla/rhino#1271] [Q&A] [answered]
Is there documentation about how to use stackTraceLimit and prepareStackTrace for a NativeError instance?

I tried the following code but nothing printed in console:

```js
Error.prepareStackTrace = function () {
    console.log(arguments);
    return "test";
};
try {
    (0).toFixed(-1);
} catch (e) {
    // Ignored.
}
```

Also, code like `Error.stackTraceLimit = 5;` seems invalid.

BTW, i found some code may have some relations.

https://github.com/mozilla/rhino/blob/4eeea69968e7fd656c4223e04dc3432cf11b9aee/src/org/mozilla/javascript/NativeError.java#L81
https://github.com/mozilla/rhino/blob/4eeea69968e7fd656c4223e04dc3432cf11b9aee/src/org/mozilla/javascript/NativeError.java#L182-L183

What makes me confused is that L183 always got a null result (for variable "pp").

**Comment by p-bakker** [comment, marked as answer]
There's no specific Rhino documentation, as it should function the same as in v8/node.

You could have a look at our testcases for these option: https://github.com/mozilla/rhino/blob/1b7e8ab9f18011756e0298812497b16c41006ff7/testsrc/jstests/extensions/stack-traces-v8.js

**Reply by SuperMonster003** [reply]
Thank you so much. :)
